### DISCUSSION TITLE
HowTo

### DISCUSSION THREAD

**Discussion by DingBEN** [pawelsalawa/sqlitestudio#3641] [Q&A]
### Details
bear with me... I am looking for a QUICK simple non disruptive enhancement
Though on simple functionality an interface should be intuitive and not require any instructions, to arrive at that intuitive level the interface needs changes.

**TEMPORARY SIMPLE SOLUTION:**
@@@ add a 'howto' at the top of the 'Help menu dropdown'
let it bring up a static page of basics, alphabetical and very topical (one-liners mostly)
yeah, the quick rebuttal is to say there is a help section in the top menu... **not the answer!**

**_I actually would put it as a new dropdown next to help so it is readily VISIBLE_**

**ONCE you have read this, see the QUESTION at the bottom**
As mentioned in the colors issue(where the need of a visual to know how one adds data is brought up), one should not have to struggle to navigate the windows.

The windows list seemed to be the only way to find the data for a table and insert key or + to add a row, but what if the windows list is empty!? or the way they are named helps not!
...and what if only SQL Editor entries are in the windows list and one has no clue the datagrid associated does not have the functionality. [ this is a blind side issue ]

as I am struggling to go to another database and see data in a table I just clickuhdyclick on everything to find my path to it.
-  and I see that doubleclicking a table brings the Data view for that table...
- and I think, ONLY because I did change the DATA tab to be first... but not sure... if last visited struct it would bring up the table structure NOT THE DATA (yes it does, just tested it)
As a new user this is so frustrating. - THIS IS NEITHER INTUITIVE NOR EVEN EXPLICIT!

**THE HOWTO would have 2 lines:**
- INSERT DATA 1 - SQL way - SQLEditor in tools (use INSERT SQL command, no VISUAL way to manipulate data)
- INSERT DATA  2  -  DOUBLE CLICK a table and click the DATA tab! use + or insert key to open a new empty row
- **3rd line: LOOOOOOKOUT!!!** the Del key is just as hot as the Insert key... if you inadvertently hit it POOOOOF! ROW is gone
- **_ADDED after the fact:_**  I just discovered the windows list can be grabbed by its handle, undocked(whether vertical or horizontal) and then dragged to the very top over the title-bar(windows) to gain vertical or horizontal real-estate... THAT should be in the HowTo!!!

The first entry above, I had used intuitively because I happen to know SQL, and never realized there was a second dataView until I lucked up on it and did not realize it took me to similarly looking data grid window with separate functionality, then lost it, and had no idea how to bring it back up!
I saw the data grids as one and the same!

... reading the lengthy prosaic Help, was a waste, some of the above not mentioned, and what I could find was buried so deep I stopped reading.

Now I know how it works, I know each time a table is double-clicked it creates an entry in the windows list, but if I have not double-clicked and am not familiar, I am stumped, I have no idea how I was able to get to the data unless I use the SQL Editor and sue SQL!

...and I do not need the windows list when I know can double-click on the table in the database explorer to get to the data window and make sure to remember to click on the DATA TAB!
- the HowTo would have made it easy

done

**Question:** Now, without going to help, see if you can think of where the double-click/insert/+ is explained - to get to a point where one can deal with the data in a table!?
...in the mass of topics in the help section, manual or documentation

and why this temporary HowTo  - until GUI is made more intuitive(or forever!)

**Comment by pawelsalawa** [comment]
I will be honest with you.

For last 3 years I barely have time to manage issues reported on GitHub. Occassionaly I have few moments to write some code and fix some issues, add some features. I heavily struggle for time to work on this project, while I would love to do so more.

Given the circumstances, I would really prefer to spend time on actual coding, bugfixes & features, instead of writing user manuals or how-to's.

Having said above, I'd like also to refer to the "until GUI is made more intuitive" statement. The UX for this app was heavily inspired by other, highly professional software for other databases (like SQL Developer for Oracle, MS SQLServer Management Studio, Toad for Oracle, EMS PostgreSQL manager).

This application is not extreamly simple, because it supports advanced features of SQLite and adds its own sophisticated features on top of it. Usage will get more complex, as there are more possibilities, more options in the application.

I'm not saying it's perfect, but at the same time, according to many feedbacks I received, it's easy to use. There are more important things on the TODO list.

**Comment by DingBEN** [comment]
**touche.** LOL  I am not sweating it... I suggest... if any of my suggestions end up being implemented to improve the tool, I am all the more satisfied, if not... I let it ride.

I feel your pain, I am a programmer, I go thru the same waves.
It gets to a point where one has to let great things lie dormant because other more pressing issues take one's time. **DO what you can... YOU ARE APPRECIATED.**

As to these great experts whom you list, to be rather blunt, I PISS on them! **Let me explain!**
I have used so many database tools (Unify, XBase(DBase etc), Profile, FilePro, Oracle, SQL Server, MySQL, BrightStar, SQLite, ... the list is long AND I am not just listing them, I HAVE PROGRAMMED IN ALL OF THEM and many more!
Guess what... the very same experts repeating the same lame approaches year after year after year!
all the way back to 1978(for me)
..and yes, the nature of the business competition is a driving factor behind having to code the least necessary to get the tools out before anyone else does.

When I attach myself to a tool, I feel obligated to try to improve it.
I am retired, won't be on Earth for much longer, and cannot spend too much time outside of my commitments... thus my help is limited to suggestions and expression of thought.
I could send you INNER/OUTER/wateva JOINS Oracle queries that would tie your neurons to your toes hairs and do not fit in a pamphlet... but not any more, I am into simple shtuff now, coasting the last 5th of the life race.

I may not send you anymore shtuff to decipher, I am pretty much done.
...and I get the message, you are but one man at the bench, I do not need to add to your dilemma.

The improvements I have proposed so far are directed at the noob, the first-time user who is basically trying to decide if the tool has ended the search for such tool. Surely we run into a contradiction here, if you are so busy, perhaps you do not want the tool to take on a trail of fire and become the de-facto tool of choice YOU will have to stand behind, while you swiftly start hiring help to satisfy the load. 

**A small example:**
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
 TRY to find in the SQL documentations of all the various kind(SQLite included) what would one use to search for a particular partial content of a column(field) ... it's a struggle, yet this is one of the earliest needs one has to deal with in querying data.
Much less what is that keyword in SQL!?  such as 'LIKE' and moreover why you would not use LIKE if CHARINDEX exists in the implementation, much less knowing these keywords(operators) even exist.
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Having said that:  ALL software I have used that blackbox the process steps by trying to guess for the user what the next context flow is, have been but painful.
NO two programmers implements this in a consistent way.
It is much easier to deal with explicit steps one can click in proper visual order than to constantly trip because the flow is implied. ...even if more navigation clicks are necessary. Having to 2nd guess oneself constantly is like walking a tight-rope.

Surely, someone who uses a tool daily and heavily covering all aspects of the functionality becomes so seasoned with the particular implementation and can fly thru it.
**BUT!!!** how many people out there have this type context and experience compare to the noob masses!  - 1% to 99% ratio!
Most of us come back to tools we use after an extended hiatus, and here we go again, how does this app like to do this!?
I nearly stopped evaluating SQLiteStudio while battling the invisible.
As I told you on previous posts, the fact that you addressed the issue fast and resolved it in good time, had me stick around... and I am making this tool the chosen one.

My aim is to have a raw tool to work with SQLite on the real back-end, raw database management.
I will not be in this tool nearly as much as I am in the web front end server side, where essentially embedded SQL is consumed via node.js and served to a web front-end.

So when I go back to the tool, I don't want to have to remember any twisted implied set of steps.
**NOTE:** as of now I have created a database 'HOWTO' with a 'howto' table and I am sliding in it everything I see will give me future clonk-moments!

Here is how I have learned how to approach any process.
The human mind has a way to get confused between what an app dictates is the focus, versus what the job dictates. So we vacillate between what is imposed(learned) IN EVERY AND EACH APP vs what is intuitive.

The main idea is to place the proper elements in the visual area and let the user flow thru that.

**Terms are critical:**
For example: You use the word ADD and remove...
- when add(has to be qualified with '_to the list_')
- and remove scares the bonbons outta anyone who has had the bad experience of watching entire data sets vanish...  **Visual Studio of Microsoft** has one of these confusions... you never know if the data will be removed from the Folder Tree in VS or zapped on disk or both!

**Words like USE, SHOW, ATTACH, DETACH, include, exclude, etc...** are much more precise and intuitive and safe, leaving little doubt as to the interpretation.
remove is too close to delete... one is forced to trip. (**UNIX REMOVE = DELETE, RMDIR = DELETE** for example)

When it comes to data, the mind thinks of data, not structure, not anything else, thus the job should be datacentric(_big complex word depending on the context_) not process-step-centric.

A button bar for each process step will quickly overwhelm the display real-estate and become a panning or scrolling proposition, a pain, or useless.  A full screen height options select does the job. A database table even better.
My earlier suggestion to have the windows list buttons be auto named was too nice... trying not to rock the boat.
I can more readily and much faster select a row from a table that would hold all these windows/queries, especially if I am not constantly tripped by a window that offers a query that does not apply to the 'CURRENT' database/table set.
By taking a table approach almost ALL options can be in the same format... same place(trigger).
One quickly triggers the system database table view, picks the context tab, and toggles back to the proper panel, no windows list needed, no real-estate wasted, and the pick(row) shows much more data to ease the selection... no need to match buttons to a hidden-set-one-must-remember, no tripping!
The history functionality almost can do that... except it too fails if the current database is a mismatch. This is where the context can move implicitly... don't go clonk because the current database is other.

A STRUCTURE trigger CAN be presented side by side as a choice, so it is visually present, but since it is not used as frequently, no need to assume that is what is desired, force a choosing as needed, rather.
...and if there is a need to have STRUCTURE as the main choice, simply allow the context to be set in a drop down(temporarily).

_An example that speaks loads, when verbs that are either 'dough!' or have nothing to do with the focus are used as the primary key in a title/sequence..._
**When one wants to post an issue, for example:**
...could title it:
- ____ 1. How to contemplate the reason a DELETE should be considered
- OR: 2. DELETE - reason considerations
- OR: 3. My employees table fails to correctly VACUUM
- OR: 4. VACUUM fails some rows
- OR: 5. How to SELECT rows and display NULLs last
- OR: 6. NULLs - SELECT nulls last
_...then someone attempts to search entries and gets a result set that is alphabetically ordered._
1, 3, and 5 are what one sees in all forums... and one has to read massive crap constantly, to find out the entry is not the desired one.
2, 4 and 6. sort nicely and are easy to scan and pick
All the forums moderators spend more time trying to tell users how to do this or that when they could simply spend time to properly re-title all posts.
Thanks to the use of TAGs some of the above is alleviated.

ok... hopefully some of the above will be of help.
